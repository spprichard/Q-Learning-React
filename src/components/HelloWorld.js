import React, { Component } from 'react';

class HelloWorld extends Component {
  constructor(props){
    super(props);
    this.frenchify = this.frenchify.bind(this)
    this.resetGreeting = this.resetGreeting.bind(this)
    this.state = {
      greeting: "Hello "
    };
    
  }

  frenchify(){
    this.setState({
      greeting: "Bonjur "
    });
  }

  resetGreeting(){
    this.setState({
      greeting: "Hello "
    });
  }

  render() {
    return (
      <div className="HelloWorld">
        <h1>{this.state.greeting} {this.props.name}!</h1>
        <br/>
        <button className="RegularButton" onClick={this.frenchify}>Frenchify!</button>
        <button className="ResetButton" onClick={this.resetGreeting}>Reset</button>
      </div>
    );
  }
}

export default HelloWorld;
