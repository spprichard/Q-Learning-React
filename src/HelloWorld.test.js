import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import HelloWorld from './components/HelloWorld';


describe(HelloWorld, () => {
    const name = 'Person';
    const mockRemoveGreeting = jest.fn();
    const mockReset = jest.fn();
    const component = shallow(
        <HelloWorld name={name} removeGreeting={mockRemoveGreeting}/>
    );

    // If somone makes changes to HelloWorld component without
    // tests to support changes, this test will break
    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <HelloWorld name="Person"/> 
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the supplied name', () => {
        expect(component.text()).toContain(name)
    });

    it("modifies the greeting when frenchify buttin is clicked", () =>{
        component.find("button.FrenchifyButton").simulate('click');
        expect(component.text()).toContain("Bonjour");
    });

    it("calls the passed in removeGreeting function when remove button is clicked", () => {
        component.find("button.RemoveButton").simulate('click');
        expect(mockRemoveGreeting).toBeCalled();
    });

});