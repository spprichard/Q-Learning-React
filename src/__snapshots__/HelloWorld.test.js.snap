// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorld extends _react.Component {
  constructor(props) {
    super(props);
    this.frenchify = this.frenchify.bind(this);
    this.resetGreeting = this.resetGreeting.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
    this.state = {
      greeting: "Hello " };


  }

  frenchify() {
    this.setState({
      greeting: "Bonjour " });

  }

  resetGreeting() {
    this.setState({
      greeting: "Hello " });

  }

  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }

  render() {
    return (
      _react2.default.createElement("div", { className: "HelloWorld", __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this },
        _react2.default.createElement("h1", { __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this }, this.state.greeting, " ", this.props.name, "!"),
        _react2.default.createElement("br", { __source: { fileName: _jsxFileName, lineNumber: 35 }, __self: this }),
        _react2.default.createElement("button", { className: "FrenchifyButton", onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 36 }, __self: this }, "Frenchify!"),
        _react2.default.createElement("button", { className: "ResetButton", onClick: this.resetGreeting, __source: { fileName: _jsxFileName, lineNumber: 37 }, __self: this }, "Reset"),
        _react2.default.createElement("button", { className: "RemoveButton", onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this }, "Remove Me")));


  }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorld"
>
  <h1>
    Hello 
     
    Person
    !
  </h1>
  <br />
  <button
    className="FrenchifyButton"
    onClick={[Function]}
  >
    Frenchify!
  </button>
  <button
    className="ResetButton"
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="RemoveButton"
    onClick={[Function]}
  >
    Remove Me
  </button>
</div>
`;
